wrangler-core/src/main/java/io/cdap/wrangler/expression/ELContext.java

```diff
@@ -45,24 +63,40 @@ public ELContext() {
 
   /**
    * Constructor that extracts the {@link ExecutorContext} internals and turns them into variables.
-   * This method extracts the trasient variables, runtime arguments, environment it's running in and
+   * This method extracts the transient variables, runtime arguments, environment it's running in and
    * the context in which it is running as identifiers that can be used within JEXL expression.
    *
    * @param context to be examined to be extracted into JEXL expression variables.
    */
   public ELContext(ExecutorContext context) {
-    if (context == null) {
-      return;
-    }
+    init(context);
+  }
 
-    // Adds the transient store variables.
-    for (String variable : context.getTransientStore().getVariables()) {
-      values.put(variable, context.getTransientStore().get(variable));
+  /**
+   * Sets the context for EL, includes the required variables in expression, 'this' and 'ctx'.
+   *
+   * @param context to be examined to be extracted into JEXL expression variables.
+   * @param el the expression.
+   * @param row the row for 'this'.
+   */
+  public ELContext(ExecutorContext context, EL el, Row row) {
+    for (String var : el.variables()) {
+      set(var, row.getValue(var));
     }
+    // These two steps below has to always happen after el variables are loaded
+    // because, el variables might not be present in a row.
+    init(context);
+    set("this", row);
+  }
 
-    values.put("runtime", context.getProperties());
-    values.put("environment", context.getEnvironment().name());
-    values.put("context", context.getContextName());
+  private void init(ExecutorContext context) {
+    if (context != null) {
```

Comment:
when can context be null? If context can be null it should be marked as nullable in caller as well as in this method

Reply:
during test

Reply:
done